packages/onboarding/src/step-container/index.tsx

```diff
@@ -0,0 +1,231 @@
+import { ActionButtons } from '@automattic/onboarding';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { ReactChild, ReactElement } from 'react';
+import StepNavigationLink from '../step-navigation-link';
+import './style.scss';
+
+interface Props {
+	headerText: string | ReactChild;
+	subHeaderText: string | ReactChild;
+	stepContent: ReactElement;
+	hideBack?: boolean;
+	hideSkip?: boolean;
+	hideNext?: boolean;
+	formattedHeader?: ReactElement;
+	shouldHideNavButtons?: boolean;
```

Comment:
To help simplify this component a little would it be possible to combine `hideBack`, `hideSkip`, `hideNext`, and `shouldHideNavButtons` into an array. Expecting either the buttons you wish to hide or `all`.

I think the same could be done for the label props. Looking at the current uses for the `NavigationLink` component and it doesn't seem to be anyone offering a different label text for back or forward.

Reply:
That would make dynamic conditions painful. Imagine `<StepContainer hideBack={ isAtomic } ... />` how would this look if it was an array?